pkgs/development/tools/build-managers/xmake/default.nix

```diff
@@ -1,48 +1,29 @@
-{ lib
-, stdenv
-, fetchurl
-, pkg-config
-, lua
-, readline
-, ncurses
-, lz4
-, tbox
-, xmake-core-sv
+{
+  lib,
+  stdenv,
+  fetchurl,
+  CoreServices,
+  ...
 }:
-
 stdenv.mkDerivation rec {
   pname = "xmake";
-  version = "2.9.4";
-
+  version = "v2.9.5";
   src = fetchurl {
-    url = "https://github.com/xmake-io/xmake/releases/download/v${version}/xmake-v${version}.tar.gz";
-    hash = "sha256-deLd4r0qSKMymJuAGuZQd8RS1JH+xRep2yeoHIcTzcU=";
+    url = "https://github.com/xmake-io/xmake/releases/download/${version}/xmake-${version}.tar.gz";
+    hash = "sha256-A/61eH4i+rjdQEGew9hKvTWrzZ+KGyTEiMfrVx1nJMg=";
   };
 
-  nativeBuildInputs = [
-    pkg-config
-  ];
-
-  buildInputs = [
-    lua
-    lua.pkgs.cjson
-    readline
-    ncurses
-    lz4
-    tbox
-    xmake-core-sv
-  ];
+  patches = [ ./0001-Fix-platform-compatibility-for-_os_find-and-_os_date.patch ];
 
-  strictDeps = true;
-
-  configureFlags = [ "--external=y" ];
+  buildInputs =
+    [
+    ]
+    ++ lib.optional stdenv.hostPlatform.isDarwin CoreServices;
 
   meta = with lib; {
     description = "Cross-platform build utility based on Lua";
     homepage = "https://xmake.io";
     license = licenses.asl20;
-    platforms = lua.meta.platforms;
     maintainers = with maintainers; [ rewine ];
```

Comment:
You can add yourself to maintainers if you want，I originally packaged xmake to build [mogan](https://github.com/NixOS/nixpkgs/issues/270803
), but the new version of mogan is more complicated, so I finally gave up

It would be good if there is an xmake user who maintains this package